web/src/routes/admin/system-settings/+page.svelte

```diff
@@ -7,55 +7,152 @@
   import PasswordLoginSettings from '$lib/components/admin-page/settings/password-login/password-login-settings.svelte';
   import SettingAccordion from '$lib/components/admin-page/settings/setting-accordion.svelte';
   import StorageTemplateSettings from '$lib/components/admin-page/settings/storage-template/storage-template-settings.svelte';
-  import LoadingSpinner from '$lib/components/shared-components/loading-spinner.svelte';
-  import { api } from '@api';
+  import { api, type SystemConfigDto, type SystemConfigTemplateStorageOptionDto } from '@api';
   import type { PageData } from './$types';
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
 
   export let data: PageData;
+  let config: SystemConfigDto = data.config;
 
-  const getConfig = async () => {
-    const { data } = await api.systemConfigApi.getConfig();
-    return data;
+  let currentConfig: SystemConfigDto = JSON.parse(JSON.stringify(config));
```

Comment:
Why do we have to parse here? isn't the data coming back is already typed?

Reply:
He needs an independent copy. If you assign it, both variables will reference the same object and both will be updated any time a property changes. This is a little hack to deep clone the object so you have two separate versions of it.

Reply:
Ah got it, deep cloning makes sense

Reply:
We already have lodash. Could use their deep clone instead.

Reply:
Yes exactly !
Thanks for the loadash suggestion. I'm using it now